Warn about outdated python in venv and live with it

```
E       LookupError: cannot choose python in /usr/bin
```

```
❯ /home/gram/.local/share/dephell/venvs/dephell-nLn6/pytest/bin/python --version
Python 3.6.7

❯ python3.6 --version
Python 3.6.8
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,7 @@
 language: python
 
 before_install:
+  - ls /opt/python/
   # show a little bit more information about environment
   - sudo apt-get install -y tree
   - env

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,11 +1,16 @@
 language: python
 
+# do not run Travis for PR's twice (as for push and as for PR)
+branches:
+  only:
+    - master
+
 before_install:
-  - ls /opt/python/
   # show a little bit more information about environment
   - sudo apt-get install -y tree
   - env
   - tree
+  - ls /opt/python/
   # install DepHell
   # https://github.com/travis-ci/travis-ci/issues/8589
   - curl https://raw.githubusercontent.com/dephell/dephell/master/install.py | /opt/python/3.6/bin/python

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,14 +24,16 @@ matrix:
   include:
     - python: "3.5"
       env: ENV=pytest
-    - python: "3.6"
+    - python: "3.6.7"
       env: ENV=pytest
-    - python: "3.7-dev"
+    - python: "3.7"
+      env: ENV=pytest
+    - python: "3.8-dev"
       env: ENV=pytest
     - python: "pypy3.5"
       env: ENV=pytest
 
-    - python: "3.6"
+    - python: "3.7"
       env: ENV=flake8
-    - python: "3.6"
+    - python: "3.7"
       env: ENV=typing

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,10 +10,9 @@ before_install:
   - sudo apt-get install -y tree
   - env
   - tree
-  - ls /opt/python/
   # install DepHell
   # https://github.com/travis-ci/travis-ci/issues/8589
-  - curl https://raw.githubusercontent.com/dephell/dephell/master/install.py | /opt/python/3.6/bin/python
+  - curl https://raw.githubusercontent.com/dephell/dephell/master/install.py | /opt/python/3.6.7/bin/python
   - dephell inspect self
 install:
   - dephell venv create --env=$ENV --python="/opt/python/$TRAVIS_PYTHON_VERSION/bin/python"

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -32,6 +32,7 @@ init:
 install:
   - "%PYTHON%/Scripts/pip.exe install pytest"
   - "%PYTHON%/Scripts/pip.exe install attrs"
+  - "%PYTHON%/Scripts/pip.exe install requests"
   - "%PYTHON%/Scripts/pip.exe install dephell-pythons"
 
 test_script:

--- a/dephell_venvs/_builder.py
+++ b/dephell_venvs/_builder.py
@@ -56,7 +56,16 @@ class VEnvBuilder(EnvBuilder):
         for path in paths:
             if finder.get_version(path) == version:
                 return str(path)
-        raise LookupError('cannot choose python in ' + str(lib_path))
+
+        # get from these pythons python with the same major.minor version
+        version = '.'.join(version.split('.')[:2])
+        for path in paths:
+            bin_version = finder.get_version(path)
+            bin_version = '.'.join(bin_version.split('.')[:2])
+            if bin_version == version:
+                return str(path)
+
+        raise LookupError('cannot choice python in ' + str(lib_path))
 
     def ensure_directories(self, env_dir: str) -> SimpleNamespace:
         context = super().ensure_directories(env_dir)

--- a/dephell_venvs/_builder.py
+++ b/dephell_venvs/_builder.py
@@ -58,11 +58,10 @@ class VEnvBuilder(EnvBuilder):
                 return str(path)
 
         # get from these pythons python with the same major.minor version
-        version = '.'.join(version.split('.')[:2])
+        major, minor, *_ = version.split('.')
         for path in paths:
-            bin_version = finder.get_version(path)
-            bin_version = '.'.join(bin_version.split('.')[:2])
-            if bin_version == version:
+            bmajor, bminor, *_ = finder.get_version(path).split('.')
+            if bmajor == major and bminor == minor:
                 return str(path)
 
         raise LookupError('cannot choice python in ' + str(lib_path))

--- a/dephell_venvs/_builder.py
+++ b/dephell_venvs/_builder.py
@@ -103,11 +103,17 @@ class VEnvBuilder(EnvBuilder):
                 dest_library.chmod(0o755)
 
     def _setup_pip(self, context: SimpleNamespace) -> None:
+        # make environment
         env = {}
         # https://github.com/appveyor/ci/issues/1995
         env.update(os.environ)
+        for name in env.copy():
+            if name.startswith('PYTHON'):
+                del env[name]
         if not native_ensurepip_exists():
             env['PYTHONPATH'] = str(get_path().parent)
+
+        # run
         result = subprocess.run(
             [context.env_exe, '-sm', 'ensurepip', '--upgrade', '--default-pip'],
             stdout=subprocess.PIPE,

--- a/dephell_venvs/_builder.py
+++ b/dephell_venvs/_builder.py
@@ -11,6 +11,8 @@ from venv import EnvBuilder
 import attr
 from dephell_pythons import Finder
 
+from ._ensurepip import get_path, native_ensurepip_exists
+
 
 @attr.s()
 class VEnvBuilder(EnvBuilder):
@@ -93,8 +95,15 @@ class VEnvBuilder(EnvBuilder):
                 dest_library.chmod(0o755)
 
     def _setup_pip(self, context: SimpleNamespace) -> None:
-        cmd = [context.env_exe, '-Im', 'ensurepip', '--upgrade', '--default-pip']
-        result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        env = {}
+        if not native_ensurepip_exists():
+            env['PYTHONPATH'] = str(get_path().parent)
+        result = subprocess.run(
+            [context.env_exe, '-sm', 'ensurepip', '--upgrade', '--default-pip'],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=env,
+        )
         if result.returncode != 0:
             output = result.stdout.decode() + '\n\n' + result.stderr.decode()
             raise OSError(output)

--- a/dephell_venvs/_builder.py
+++ b/dephell_venvs/_builder.py
@@ -104,6 +104,8 @@ class VEnvBuilder(EnvBuilder):
 
     def _setup_pip(self, context: SimpleNamespace) -> None:
         env = {}
+        # https://github.com/appveyor/ci/issues/1995
+        env.update(os.environ)
         if not native_ensurepip_exists():
             env['PYTHONPATH'] = str(get_path().parent)
         result = subprocess.run(

--- a/dephell_venvs/_builder.py
+++ b/dephell_venvs/_builder.py
@@ -4,7 +4,7 @@ import subprocess
 import sys
 from pathlib import Path
 from types import SimpleNamespace
-from typing import Optional
+from typing import Dict, Optional
 from venv import EnvBuilder
 
 # external
@@ -104,7 +104,7 @@ class VEnvBuilder(EnvBuilder):
 
     def _setup_pip(self, context: SimpleNamespace) -> None:
         # make environment
-        env = {}
+        env = {}    # type: Dict[str, str]
         # https://github.com/appveyor/ci/issues/1995
         env.update(os.environ)
         for name in env.copy():

--- a/dephell_venvs/_ensurepip.py
+++ b/dephell_venvs/_ensurepip.py
@@ -1,0 +1,47 @@
+from pathlib import Path
+from typing import Iterator
+
+import requests
+
+
+VERSION = '3.6'
+URLS = (
+    'https://api.github.com/repos/python/cpython/contents/Lib/ensurepip?ref=',
+    'https://api.github.com/repos/python/cpython/contents/Lib/ensurepip/_bundled?ref=',
+)
+
+
+def get_links() -> Iterator[str]:
+    for url in URLS:
+        url += VERSION
+        response = requests.get(url)
+        response.raise_for_status()
+        for info in response.json():
+            if info['type'] == 'file':
+                yield info['download_url']
+
+
+def download_files(path: Path) -> None:
+    for link in get_links():
+        relpath = link.split('/ensurepip/', maxsplit=1)[1]
+        fpath = path.joinpath(relpath)
+        fpath.parent.mkdir(exist_ok=True, parents=True)
+        response = requests.get(link)
+        response.raise_for_status()
+        fpath.write_bytes(response.content)
+
+
+def get_path() -> Path:
+    path = Path(__file__).absolute().parent / 'ensurepip'
+    if not path.exists():
+        download_files(path)
+    return path
+
+
+def native_ensurepip_exists() -> bool:
+    try:
+        import ensurepip  # noqa
+    except ImportError:
+        return False
+    else:
+        return True

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -52,4 +52,4 @@ classifiers = [
 python = ">=3.5"
 attrs = "*"
 dephell-pythons = "*"
-
+requests = "*"

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -11,3 +11,11 @@ def temp_path(tmp_path: Path):
         else:
             shutil.rmtree(str(path))
     yield tmp_path
+
+
+@pytest.fixture(autouse=True)
+def drop_ensurepip():
+    yield
+    path = Path(__file__).parent.parent / 'dephell_venvs' / 'ensurepip'
+    if path.exists():
+        shutil.rmtree(str(path))

